contracts/vaults/STETHVault/RibbonThetaSTETHVault.sol

```diff
@@ -0,0 +1,356 @@
+        optionState.nextOption = otokenAddress;
+        optionState.nextOptionReadyAt = uint32(block.timestamp.add(delay));
+
+        _closeShort(oldOption);
+    }
+
+    /**
+     * @notice Closes the existing short position for the vault.
+     */
+    function _closeShort(address oldOption) private {
+        optionState.currentOption = address(0);
+
+        uint104 lockedAmount = vaultState.lockedAmount;
+        vaultState.lastLockedAmount = lockedAmount > 0
+            ? lockedAmount
+            : vaultState.lastLockedAmount;
+        vaultState.lockedAmount = 0;
+
+        if (oldOption != address(0)) {
+            uint256 withdrawAmount =
+                VaultLifecycleSTETH.settleShort(GAMMA_CONTROLLER);
+            emit CloseShort(oldOption, withdrawAmount, msg.sender);
+        }
+    }
+
+    /**
+     * @notice Rolls the vault's funds into a new short position.
+     */
+    function rollToNextOption() external onlyKeeper nonReentrant {
+        (address newOption, uint256 queuedWithdrawAmount) = _rollToNextOption();
+
+        // Locked balance denominated in `collateralToken`
+
+        uint256 lockedBalance =
+            collateralToken.balanceOf(address(this)).sub(
+                collateralToken.getWstETHByStETH(queuedWithdrawAmount)
+            );
+
+        emit OpenShort(newOption, lockedBalance, msg.sender);
+
+        VaultLifecycleSTETH.createShort(
+            GAMMA_CONTROLLER,
+            MARGIN_POOL,
+            newOption,
+            lockedBalance
+        );
+
+        startAuction();
+    }
+
+    /**
+     * @notice Initiate the gnosis auction.
+     */
+    function startAuction() public onlyKeeper {
+        GnosisAuction.AuctionDetails memory auctionDetails;
+
+        uint256 currOtokenPremium = currentOtokenPremium;
+
+        require(currOtokenPremium > 0, "!currentOtokenPremium");
+
+        auctionDetails.oTokenAddress = optionState.currentOption;
+        auctionDetails.gnosisEasyAuction = GNOSIS_EASY_AUCTION;
+        auctionDetails.asset = vaultParams.asset;
+        auctionDetails.assetDecimals = vaultParams.decimals;
+        auctionDetails.oTokenPremium = currOtokenPremium;
+        auctionDetails.duration = auctionDuration;
+
+        optionAuctionID = VaultLifecycleSTETH.startAuction(auctionDetails);
+    }
+
+    /**
+     * @notice Burn the remaining oTokens left over from gnosis auction.
+     */
+    function burnRemainingOTokens() external onlyKeeper nonReentrant {
+        uint256 unlockedAssedAmount =
+            VaultLifecycleSTETH.burnOtokens(
+                GAMMA_CONTROLLER,
+                optionState.currentOption
+            );
+
+        if (unlockedAssedAmount > 0) {
+            vaultState.lockedAmount = uint104(
+                uint256(vaultState.lockedAmount).sub(unlockedAssedAmount)
+            );
+        }
+
+        // Wrap entire `asset` balance to `collateralToken` balance
+        VaultLifecycleSTETH.wrapToYieldToken(WETH, address(collateralToken));
```

Comment:
Why do we have to `wrapToYieldToken` if we are receiving wstETH after burning?

Reply:
for the same reason we do it in yearn. 

burnRemainingOTokens() is one of the functions we 100% know is called after the auction, hence after we received premiums for this week. so if we do end up having to burn otokens, why not maximize efficiency and wrap the premium to yield bearing token right then and there? otherwise it lays idle for a week.

Reply:
Oh I see - I did not consider the premiums that will be in WETH.